src/sdl2/render.rs

```diff
@@ -250,8 +251,33 @@ impl Renderer {
         }
     }
 
+    /// Accesses the Window properties, such as the position, size and title of a Window.
+    /// Returns None if the renderer is not associated with a Window.
+    ///
+    /// Safely calls `Window::properties()`.
+    pub fn window_properties<'b, F, R>(&'b self, _event: &'b EventPump, callback: F) -> Option<R>
+    where F: FnOnce(&mut WindowProperties<'b>) -> R
+    {
+        use std::mem;
+
+        // TODO: check if borrowed first.
+        // if it is, provide a clearer error message.
+        // needs to wait until `RefCell::borrow_state()` is stable.
+        let _borrow = self.drawer_borrow.borrow_mut();
+
+        let parent: &mut _ = unsafe { mem::transmute(&self.parent) };
```

Comment:
This is undefined behavior. See https://github.com/PistonDevelopers/piston/issues/866